src/WebJobs.Script.WebHost/Security/KeyManagement/BaseSecretsRepository.cs

```diff
@@ -14,30 +13,37 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
     public abstract class BaseSecretsRepository : ISecretsRepository, IDisposable
     {
-        private readonly AutoRecoveringFileSystemWatcher _sentinelFileWatcher;
+        private const int SentinelWatcherInitializationDelayMS = 3 * 1000;
```

Comment:
Can we make this 5 sec as our upper bound for cold start when extra dependencies are brought by function code? Also this only is needed for Consumption plan. So no need to delay this for AppService plan or for the warm instances of premium if we can detect that.